ocean_lib/ocean/ocean.py

```diff
@@ -366,3 +368,25 @@ def retrieve_provider_fees(
             provider_fees["validUntil"],
             provider_fees["providerData"],
         )
+
+    @enforce_types
+    def retrieve_provider_fees_for_compute(
+        self,
+        datasets: List[ComputeInput],
+        algorithm_data: Union[ComputeInput, AlgorithmMetadata],
+        consumer_address: str,
+        compute_environment: str,
+        valid_until: int,
+    ) -> tuple:
+
+        initialize_compute_response = DataServiceProvider.initialize_compute(
+            [x.as_dictionary() for x in datasets],
+            algorithm_data.as_dictionary(),
+            datasets[0].service.service_endpoint,
+            consumer_address,
+            compute_environment,
+            valid_until,
+        )
+
+        # TODO: maybe we should adjust this in some way?
```

Comment:
I don't think we should adjust. 

I remember when we removed the response filtering for compute status. It's a similar situation here. I think it's better for the user to get the full response.

And now that I'm glancing at retrieve_provider_fees, I'm thinking maybe we shouldn't be adjusting that response either.